### PR TITLE
Add a QueryTranslationTest command parameter to specify tests to execute

### PR DESCRIPTION
**Pull request by spena**
### Description 
The new parameter accepts a list of comma separated JSON test files to execute. This parameter allows developers to specify their own tests to execute and make debugging faster.

Example:
Execute a single test called 'sum.json'
`mvn test -pl ksql-engine -Dtest=QueryTranslationTest -Dksql.test.files=sum.json`

Execute multiple tests called 'sum.json' and 'substring.json'
`mvn test -pl ksql-engine -Dtest=QueryTranslationTest -Dksql.test.files=sum.json,substring.json`

Execute all tests:
`mvn test -pl ksql-engine -Dtest=QueryTranslationTest`

The same parameter works for `SchemaTranslationTest`.
### Testing done 
The same QueryTranslationTest and SchemaTranslationTest validate that all tests are executed.
I also executed the example commands manually to verify they are working correctly.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java [hunk 1]
```diff
@@ -779,10 +786,42 @@ private static TopologyAndConfigs readTopologyFile(final String file, final Obje
     }
   }
 
-  static Stream<JsonTestCase> findTestCases(final Path dir) {
+  private static List<Path> getTests(final Path dir, final List<String> files) {
+    List<Path> filePaths = new ArrayList<>();
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java]: Could this be:
`return files.stream().map(name -> dir.resolve(name.trim())).collect(Collectors.toList());`

File: ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java [hunk 2]
```diff
@@ -779,10 +786,42 @@ private static TopologyAndConfigs readTopologyFile(final String file, final Obje
...
+    }
+
+    return filePaths;
+  }
+
+  /**
+   * Returns a list of files specified in the system property 'ksql.test.file'.
+   * The list may be specified as a comma-separated string.
+   */
+  public static List<String> getTestFilesParam() {
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java]: can be package private

File: ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java [hunk 3]
```diff
@@ -779,10 +786,42 @@ private static TopologyAndConfigs readTopologyFile(final String file, final Obje
...
+    String propValue = System.getProperty(KSQL_TEST_FILES);
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java]: maybe: `return Arrays.asList(System.getProperty(KSQL_TEST_FILES, "").trim().split(","));`

File: ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java [hunk 4]
```diff
@@ -779,10 +786,42 @@ private static TopologyAndConfigs readTopologyFile(final String file, final Obje
...
+    }
+
+    return Arrays.asList(propValue.trim().split(","));
+  }
+
+  static Stream<JsonTestCase> findTestCases(final Path dir, final List<String> files) {
     final ClassLoader classLoader = EndToEndEngineTestUtil.class.getClassLoader();
 
-    return findTests(dir).stream()
+    List<Path> testPaths;
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java]: nit: i'd probably put this into a method `getTestPaths(..)`
and then do `return getTestPaths(files).stream()...` or similar

File: ksql-engine/src/test/java/io/confluent/ksql/QueryTranslationTest.java [hunk 5]
```diff
@@ -148,7 +159,9 @@ public void shouldBuildAndExecuteQueries() {
   }
 
   static Stream<TestCase> buildTestCases() {
-    return EndToEndEngineTestUtil.findTestCases(QUERY_VALIDATION_TEST_DIR)
+    List<String> testFiles = EndToEndEngineTestUtil.getTestFilesParam();
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/QueryTranslationTest.java]: nit: `final` and elsewhere

File: ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java (hunk 1 above)
Comment (spena) [reply ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java]: Thanks. Done.

File: ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java (hunk 2 above)
Comment (spena) [reply ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java]: Done.

File: ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java (hunk 3 above)
Comment (spena) [reply ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java]: That line will return an array of one empty item if ksql.test.files is empty. I'd rather return an empty array with zero elements. I simplify the code a little bit, though, to take advantage of the getProperty(var, default). Thanks.

File: ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java (hunk 4 above)
Comment (spena) [reply ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java]: Done. It made the code more clear.

File: ksql-engine/src/test/java/io/confluent/ksql/QueryTranslationTest.java (hunk 5 above)
Comment (spena) [reply ksql-engine/src/test/java/io/confluent/ksql/QueryTranslationTest.java]: Done.

**Pull request merged**